### DISCUSSION TITLE
How to set ActiveModel fields all at once?

### DISCUSSION THREAD

**Discussion by zxuqian** [SeaQL/sea-orm#734] [Q&A] [answered]
I'm using SeaOrm with Rocket. Here is the entity:

```rust
pub struct Model {
    #[sea_orm(primary_key)]
    #[serde(skip_deserializing)]
    pub id: i32,
    pub title: String,
    pub body: String,
}
```

Rocket route handler:

```rust
pub use entity::post as blog_post;
pub use entity::post::Entity as Post;

#[post("/", format = "json", data = "<blog>")]
async fn create(
    conn: Connection<'_, Db>,
    blog: Json<blog_post::Model>,
) -> (Status, Json<blog_post::Model>) {
    println!("{:?}", blog);

    let db = conn.into_inner();

    let new_blog = blog_post::ActiveModel {
        title: Set(blog.title.to_owned()),
        body: Set(blog.body.to_owned()),
        ..Default::default()
    }
    .insert(db)
    .await
    .expect("Failed to insert post");

    (Status::Created, Json(new_blog))
}
```

Here I can get `blog: Json<blog_post::Model>` directly from Rocket route, but I still have to set its fields into `blog_post::ActiveModel`, one by one, even they have the same fields:

```rust
blog_post::ActiveModel {
    title: Set(blog.title.to_owned()),
    body: Set(blog.body.to_owned()),
    ..Default::default()
}
```

Is there any way to copy all `blog` fields into `blog_post::ActiveModel`  instead of manually setting each one? Or convert the `blog: Json<blog_post::Model>` into a ready-to-insert ActiveModel?

It gets inconvenient when there are a lot of fields. 

Thanks!

**Comment by ikrivosheev** [comment]
Maybe this can help: https://github.com/SeaQL/sea-orm/pull/725?

**Comment by billy1624** [comment, marked as answer]
Hey @zxuqian, sorry for the delay. You can set the values of any ActiveModel from a JSON. For example,

```rs
let fruit = fruit::ActiveModel::from_json(json!({
    "name": "Apple",
}))?;

assert_eq!(
    fruit,
    fruit::ActiveModel {
        id: ActiveValue::NotSet,
        name: ActiveValue::Set("Apple".to_owned()),
        cake_id: ActiveValue::NotSet,
    }
);
```

https://www.sea-ql.org/SeaORM/docs/basic-crud/insert#set-activemodel-from-json-value

**Reply by rgreinho** [reply]
Can you provide an example in the case where I have a basket of fruits? 

It looks like the `fruit::ActiveModel::from_json` function cannot take a json representing a list of fruits.

What I am trying to do is something like:

```rs
let fruit_basket : Vec<fruit::ActiveModel> = fruit::ActiveModel::from_json(json!([ {"name": "Apple"},{"name": "Pear"}]))?;
```

**Reply by rgreinho** [reply]
Also, from the tests (https://github.com/SeaQL/sea-orm/blob/0.9.0/src/entity/active_model.rs#L927-L956=) it looks like in one case you can deserialize a partial json, in another case it does not work. (the test for the cake entity is expected to fail, but the test for the fruit entity is expected to pass). Could you clarify why?

**Reply by billy1624** [reply]
Hey @rgreinho, yes, you'd need to create a list of `ActiveModel`s then pass it on to `insert_many`.
https://www.sea-ql.org/SeaORM/docs/basic-crud/insert/#insert-many

**Reply by billy1624** [reply]
Because primary key field of `cake::Model` doesn't annotated with `#[serde(skip_deserializing)]`. However, primary key of `fruit::Model` does.

```rs
#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "fruit")]
pub struct Model {
    #[sea_orm(primary_key)]
    #[serde(skip_deserializing)] // Skip deserializing
    pub id: i32,
    pub name: String,
    pub cake_id: Option<i32>,
}
```
https://www.sea-ql.org/SeaORM/docs/basic-crud/insert/#set-activemodel-from-json-value
